@navikt/core/react/src/accordion/AccordionHeader.tsx

```diff
@@ -18,43 +19,40 @@ export type AccordionHeaderType = React.ForwardRefExoticComponent<
 
 const AccordionHeader: AccordionHeaderType = forwardRef(
   ({ children, className, onClick, ...rest }, ref) => {
-    const context = useContext(AccordionItemContext);
+    const itemContext = useContext(AccordionItemContext);
+    const accordionContext = useContext(AccordionContext);
 
-    if (context === null) {
+    if (accordionContext === null || itemContext === null) {
       console.error(
-        "<Accordion.Header> has to be used within an <Accordion.Item>"
+        "<Accordion.Header> has to be used within an <Accordion.Item>, which in turn must be within an <Accordion>"
       );
       return null;
     }
 
-    const handleClick = (
-      e: React.MouseEvent<HTMLButtonElement, MouseEvent>
-    ) => {
-      context.toggleOpen();
-      onClick && onClick(e);
-    };
-
     return (
       <button
-        {...rest}
         ref={ref}
+        {...rest}
         className={cl("navds-accordion__header", className)}
-        type="button"
-        onClick={handleClick}
-        aria-expanded={context.open}
+        onClick={(e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
+          itemContext.toggleOpen();
+          onClick && onClick(e);
+        }}
+        aria-expanded={itemContext.open}
```

Comment:
Tror type button må være med for å unngå feil ved bruk i <form>
```suggestion
        aria-expanded={itemContext.open}
        type="button"
```

Reply:
ved bruk i forms?

Reply:
Ikke noe vi oppmuntrer til, men er gjengangere at komponentene blir brukt inne i `<form>` (eks https://nav-it.slack.com/archives/C7NE7A8UF/p1683014658766199).